exopy_hqc_legacy/tasks/tasks/instr/transfer_awg_file_task.py

```diff
@@ -0,0 +1,484 @@
+            if n_loops > 0:
+                for ii, name_parameter in enumerate(name_parameters):
+                    self.write_in_database(name_parameter, loop_value[ii])
+            for k, v in self.sequence_vars.items():
+                seq.external_vars[k] = self.format_and_eval_string(v)
+            context.sequence_name = '{}'.format(nn+first_index)
+            compilation = context.compile_loop(seq, for_file=True,
+                                               factor=False)
+            if len(compilation) == 5:
+                res, byteseq, repeat, infos, errors = compilation
+            else:
+                print(compilation)
+            already_added = {}
+            is_first_ch = True
+            for ch_id in self.driver.defined_channels:
+                if ch_id in byteseq:
+                    used_pos = []
+                    for pos, waveform in enumerate(byteseq[ch_id]):
+                        addr = id(waveform)
+                        if addr not in already_added:
+                            seq_name_transfered = context.sequence_name +\
+                                '_Ch{}'.format(ch_id) +\
+                                '_' + str(pos)
+                            packed_waveforms[seq_name_transfered] = waveform
+                            already_added[addr] = seq_name_transfered
+                        else:
+                            seq_name_transfered = already_added[addr]
+
+                        if ch_id not in list(wfname_l.keys()):
+                            wfname_l[ch_id] = [seq_name_transfered]
+                        else:
+                            wfname_l[ch_id].append(seq_name_transfered)
+
+                        if (pos not in used_pos) and is_first_ch:
+                            nrep.append(repeat[pos])
+                            if wait_trigger and used_pos == []:
+                                trig_wait.append(1)
+                            else:
+                                trig_wait.append(0)
+                            goto_state.append(0)
+                            if start_with_event:
+                                jump_to.append(1)
+                            else:
+                                jump_to.append(0)
+                            used_pos.append(pos)
+                    is_first_ch = False
+
+        if start_with_event:
+            trig_wait = [0] + trig_wait
+            goto_state = [0] + goto_state
+            jump_to = [2] + jump_to
+            nrep = [0] + nrep
+            for ch_id in byteseq:
+                seq_name_standby = 'Standby_Ch{}'.format(ch_id)
+                wfname_l[ch_id] = [seq_name_standby] + wfname_l[ch_id]
+                packed_waveforms[seq_name_standby] = np.ones(256, dtype=np.uint16)*(2**13)
+            goto_state[-1] = 2
+        else:
+            goto_state[-1] = 1
+
+        if not res:
+            raise Exception('Failed to compile sequence')
```

Comment:
Could a slightly more specific error (RuntimeError) and the errors dictionary should be formated and attached otherwise tracking the issue is impossible.